### PR TITLE
Reorganzing the WaterTAP Costing Package

### PR DESCRIPTION
**Pull request by bknueven**
## Summary/Motivation:
The WaterTAP costing package was too big. This breaks out all its costing methods and associated parameters into their own files.

## Changes proposed in this PR:
- [x] Separate out costing methods and associated parameters from the watertap costing package into their own modules
- [x] Add basic utility costing methods and helpers to `watertap/costing/util.py`
- [x] Test associated lazy construction techniques for robustness

## Questions:
- Are the costing methods in the "right place"? We could put them closer to or even in the unit model modules.
- Does the interface make sense?
   - Here is a simple example: https://github.com/watertap-org/watertap/blob/17c4acc53133ec289af36033bf592e1bec76e3e0/watertap/costing/units/energy_recovery_device.py
   - This is a complex example: https://github.com/watertap-org/watertap/blob/17c4acc53133ec289af36033bf592e1bec76e3e0/watertap/costing/units/ion_exchange.py

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/unit_models/tests/test_ion_exchange_0D.py [hunk 1]
```diff
@@ -930,13 +930,13 @@ def obj_rule(m):
         results = solver.solve(m, tee=False)
         assert_optimal_termination(results)
 
-        assert pytest.approx(440372.278, rel=1e-5) == value(
+        assert pytest.approx(453807.598, rel=1e-5) == value(
```
Comment (TimBartholomew) [inline watertap/unit_models/tests/test_ion_exchange_0D.py:928]: Why are the ion exchange costs changing? Was a bug fixed?

**Review by TimBartholomew — APPROVED** [review]
LGTM - one minor comment

File: watertap/unit_models/tests/test_ion_exchange_0D.py (hunk 1 above)
Comment (bknueven) [reply watertap/unit_models/tests/test_ion_exchange_0D.py:928]: Yes, look at f3473c3. There were some uncaught issue with units.

**Comment by TimBartholomew** [comment]
@OOAmusat @lbibl this seems like a big PR, however, it is essentially moving the costing methods for each unit out of the WaterTAP costing package into a separate file. Could one of you review this, you don't need to go through it line by line. Just see one example of the unit model costing methods now interacts with the costing package

**Review by OOAmusat — APPROVED** [review]
LGTM

**Review by lbibl — APPROVED** [review]
Looks good and a better organization.

**Pull request merged**